Source/WebCore/platform/audio/SharedAudioDestination.cpp

```diff
@@ -61,31 +61,31 @@ class SharedAudioDestinationAdapter : public ThreadSafeRefCountedAndCanMakeThrea
     void isPlayingDidChange() final { }
 
     void configureRenderThread(CompletionHandler<void(bool)>&&);
+    void callAllConfigurationHandlers(bool);
 
     Ref<AudioDestination> protectedDestination() { return m_destination; }
     Ref<AudioBus> protectedWorkBus() { return m_workBus; }
-    Ref<WorkQueue> protectedConfigurationQueue() { return m_configurationQueue; }
 
     unsigned m_numberOfOutputChannels;
     float m_sampleRate;
 
     Ref<AudioDestination> m_destination;
     Ref<AudioBus> m_workBus;
-    Ref<WorkQueue> m_configurationQueue;
 
     bool m_started { false };
 
     Lock m_renderLock;
-    Semaphore m_configurationSemaphore;
 
-    // Only accessed on m_configurationQueue:
-    Vector<RefPtr<SharedAudioDestination>> m_renderers;
+    using RenderVector = Vector<RefPtr<SharedAudioDestination>>;
+    RenderVector m_renderers;
 
     bool m_needsConfiguration WTF_GUARDED_BY_LOCK(m_renderLock) { true };
-    Vector<RefPtr<SharedAudioDestination>> m_newRenderers WTF_GUARDED_BY_LOCK(m_renderLock);
+    RenderVector m_newRenderers WTF_GUARDED_BY_LOCK(m_renderLock);
+    using ConfigurationHandlerVector = Vector<CompletionHandler<void(bool)>>;
+    ConfigurationHandlerVector m_configurationCompletionHandlers WTF_GUARDED_BY_LOCK(m_renderLock);
 
     // Only accessed on the audio thread:
-    Vector<RefPtr<SharedAudioDestination>> m_configuredRenderers;
+    RenderVector m_configuredRenderers;
```

Comment:
can we have related `WTF_GUARDED_BY_CAPABILITY` for it then?

Reply:
I don't think there's a way to say `WTF_GUARDED_BY_CAPABILITY(!mainThread);`. At least, I don't see any use of it like that.

Reply:
Do you have access to the audio thread? then assertIsCurrent(audioThread()) 

anyway, if it's on the audio thread, we're now always accessing it while copy the m_newRenderers ; a lock in the audio thread is usually a worry.